### ISSUE TITLE
Add support for configuring Dask distributed

### ISSUE THREAD

**Issue by bouweandela** [ESMValGroup/ESMValCore#2040]
Since [iris 3.6](https://scitools-iris.readthedocs.io/en/stable/whatsnew/3.6.html), it is possible to use [Dask distributed](https://distributed.dask.org/en/stable/) with iris. This is a great new feature that will allow for better memory handling and distributed computing. See #1714 for an example implementation. However, it does require some extra configuration. 

My proposal would be to allow users to specify the arguments to [distributed.Client](https://distributed.dask.org/en/stable/api.html#client) and to the associated cluster, e.g. [distributed.LocalCluster](https://distributed.dask.org/en/stable/api.html#distributed.LocalCluster) or [dask_jobqueue.SLURMCluster](https://jobqueue.dask.org/en/latest/generated/dask_jobqueue.SLURMCluster.html#dask_jobqueue.SLURMCluster) in this configuration. This could either be added under a new key in config-user.yml or in a new configuration file in the `~/.esmvaltool` directory:

Add to the user configuration file
----------------------------------

We could add these new options to `config-user.yml` under a new key `dask`, e.g.

Example config-user.yml settings for running locally using a LocalCluster:
```yaml
dask:
  cluster:
    type: distributed.LocalCluster
```
Example settings for using an externally managed cluster (e.g. set it up from a Jupyter notebook)
```yaml
dask:
  client:
    address: tcp://127.0.0.1:45695
```
Example settings for running on Levante:
```yaml
dask:
  client: {}
  cluster:
    type: dask_jobqueue.SLURMCluster
    queue: interactive
    account: bk1088
    cores: 8
    memory: 16GiB
    local_directory: "/work/bd0854/b381141/dask-tmp"
    n_workers: 2
```

New configuration file
----------------------

Or, we could add the new configuration in a separate file, e.g. called `~/.esmvaltool/dask.yml` or `~/.esmvaltool/dask-distributed.yml`.

Example config-user.yml settings for running locally using a LocalCluster:
```yaml
cluster:
  type: distributed.LocalCluster
```
Example settings for using an externally managed cluster (e.g. set it up from a Jupyter notebook)
```yaml
client:
  address: tcp://127.0.0.1:45695
```
Example settings for running on Levante:
```yaml
client: {}
cluster:
  type: dask_jobqueue.SLURMCluster
  queue: interactive
  account: bk1088
  cores: 8
  memory: 16GiB
  local_directory: "/work/bd0854/b381141/dask-tmp"
  n_workers: 2
```

@ESMValGroup/esmvaltool-coreteam Does anyone have an opinion on what the best approach is here? A new file or add to config-user.yml?

**Comment by bouweandela** [comment]
Another question: would we like to be able to configure Dask distributed from the command line? Or at least pass in the scheduler address if we already have a Dask cluster running, e.g. started from a Jupyter notebook?

**Comment by valeriupredoi** [comment]
This is still an ongoing discussion so needs reopening

**Comment by bouweandela** [comment]
Suggestion by @sloosvel: 

> Regarding the configuration, is it possible to have multiple configurations in dask.yml? Not every recipe will require the same type of resources.

**Comment by bouweandela** [comment]
At the workshop at SMHI agreement was reached that a new configuration file format would be acceptable. I will make a proposal, but this will not be implemented in time for v2.10.
